compiler-core/src/language_server/tests/action.rs

```diff
@@ -158,11 +169,124 @@ pub fn main() {
 }
 ";
     assert_eq!(
-        remove_unused_action(code, 2),
+        apply_first_code_action_with_title(code, 2, REMOVE_UNUSED_IMPORTS_TITLE),
         expected.replace("%SPACE%", " ")
     )
 }
 
+#[test]
+fn test_remove_redundant_tuple_in_case_subject_simple() {
+    let code = "
+pub fn main() {
+  case #(1) { #(a) -> 0 }
+  case #(1, 2) { #(a, b) -> 0 }
+}
+";
+
+    let expected = "
+pub fn main() {
+  case 1 { a -> 0 }
+  case 1, 2 { a, b -> 0 }
+}
+";
+
+    assert_eq!(
+        apply_first_code_action_with_title(code, 7, REMOVE_REDUNDANT_TUPLES),
+        expected
+    );
+}
+
+#[test]
+fn test_remove_redundant_tuple_in_case_subject_nested() {
+    let code = "
+pub fn main() {
+  case #(case #(0) { #(a) -> 0 }) { #(b) -> 0 }
+}
+";
+
+    let expected = "
+pub fn main() {
+  case case 0 { a -> 0 } { b -> 0 }
+}
+";
+
+    assert_eq!(
+        apply_first_code_action_with_title(code, 7, REMOVE_REDUNDANT_TUPLES),
+        expected
+    );
+}
+
+#[test]
+fn test_remove_redundant_tuple_in_case_retain_extras() {
+    let code = "
+pub fn main() {
+  case
+    #(
+      // first comment
+      1,
+      // second comment
+      2,
+      3 // third comment before comma
+
+      ,
+
+      // fourth comment after comma
+
+    )
+  {
+    #(
+      // first comment
+      a,
+      // second comment
+      b,
+      c // third comment before comma
+
+      ,
+
+      // fourth comment after comma
+
+    ) -> 0
+  }
+}
+";
+
+    let result = apply_first_code_action_with_title(code, 20, REMOVE_REDUNDANT_TUPLES);
+
+    insta::assert_snapshot!(result);
+}
+
+#[test]
+fn test_remove_redundant_tuple_in_case_subject_ignore_ineligible() {
+    let code = "
+pub fn main() {
+  case #() { #() -> 0 }
+  case #(0), #(1) { #(a), b -> 0 }
+  case #(0), #(1) {
+    #(1), #(b) -> 0
+    a, #(0) -> 1
+    #(a), #(b) -> 2
+  }
+}
+";
```

Comment:
How come there are three case in this test? Can we make this test smaller to make it clearer what is being tested?